pkg/build/builder/cmd/scmauth/password.go

```diff
@@ -86,7 +87,12 @@ func (u UsernamePassword) Setup(baseDir string) error {
 	fmt.Fprintf(gitconfig, UserPassGitConfig, gitcredentials.Name())
 	fmt.Fprintf(gitcredentials, "%s", usernamePasswordURL.String())
 
-	return ensureGitConfigIncludes(gitconfig.Name())
+	sourceURL := u.SourceURL
+	sourceURL.User = nil
```

Comment:
I guess that's ok... just trying to anticipate advanced use cases where the user provided a gitconfig of their own, and might be counting on the username match to select the right credential